tests/pcap/src/handshake_message.rs

```diff
@@ -0,0 +1,179 @@
+            frame_num,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Builder {
+    message_type: Option<u8>,
+    capture_file: Option<String>,
+}
+
+impl Builder {
+    const TCP_PAYLOAD: &'static str = "tcp.payload";
+    const FRAME_NUM: &'static str = "frame.number";
+    const MESSAGE_TYPE: &'static str = "tls.handshake.type";
+
+    pub(crate) fn set_type(&mut self, message_type: u8) -> &mut Self {
+        self.message_type = Some(message_type);
+        self
+    }
+
+    pub fn set_capture_file(&mut self, file: &str) -> &mut Self {
+        self.capture_file = Some(file.to_string());
+        self
+    }
+
+    fn build_from_capture(self, capture: rtshark::RTSharkBuilderReady) -> Result<Vec<Message>> {
+        let tcp_capture = capture
+            .display_filter(Self::TCP_PAYLOAD)
+            .metadata_whitelist(Self::TCP_PAYLOAD)
+            .metadata_whitelist(Self::FRAME_NUM)
+            .spawn()?;
+        let payloads = read_all(tcp_capture);
+
+        let filter = if let Some(message_type) = self.message_type {
+            format!("{} == {}", Self::MESSAGE_TYPE, message_type)
+        } else {
+            Self::MESSAGE_TYPE.to_string()
+        };
+        let message_capture = capture.display_filter(&filter).spawn()?;
+
+        let mut messages = Vec::new();
+        for packet in read_all(message_capture) {
+            let frame_num = get_metadata(&packet, Builder::FRAME_NUM)
+                .context("Missing frame number")?
+                .to_string();
+
+            let context_msg = format!("Failed to parse frame {}", &frame_num);
+            let message =
+                Message::from_packet(packet, frame_num, &payloads).context(context_msg)?;
+
+            messages.push(message);
+        }
+        Ok(messages)
+    }
+
+    pub(crate) fn build(mut self) -> Result<Vec<Message>> {
+        let file = self
+            .capture_file
+            .take()
+            .context("No capture file provided")?;
+        let capture = RTSharkBuilder::builder().input_path(&file);
+        self.build_from_capture(capture)
+            .with_context(|| format!("Failed to parse capture file {}", &file))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn fragmentation() -> Result<()> {
+        let mut builder = Builder::default();
+        builder.set_capture_file("data/fragmented_ch.pcap");
+        let messages = builder.build()?;
+
+        let first = messages.first().unwrap();
+        assert_eq!(first.payloads.len(), 3);
+        let bytes = first.to_bytes()?;
+        // Correct length read from wireshark
+        assert_eq!(bytes.len(), 16262);
+
+        Ok(())
+    }
+
+    #[test]
+    fn multiple_handshakes() -> Result<()> {
+        let mut builder = Builder::default();
+        builder.set_capture_file("data/multiple_hellos.pcap");
+        let messages = builder.build()?;
+        let count = messages.iter().filter(|m| m.message_type == 1).count();
```

Comment:
I think we should make `message_type` an enum, because I am struggling to understand what's happening here. I'm sure I could figure out what `message_type` means if I go back and look at the RFC, but even just having an example of a `message_type` would be appreciated 😄